compression/lpf.py

```diff
@@ -0,0 +1,73 @@
+
+from string_indexing import lcp, suffix_array
+
+
+def naive(w, n):
+  ret = []
+  for i in range(1, n + 1):
+    right = w[i:]
+    while True:
+      if right in w[1:(i + len(right) - 1)]:
+        ret += [len(right)]
+        break
+      right = right[:-1]
+  return [-1] + ret
+
+
+def chrochemore_ille_smith(text, n, SA=None, LCP=None):
+  SA = SA or suffix_array.small_large(text, n)
+  LCP = LCP or lcp.kasai(SA, text, n)
+  return compute_cis_lpf(n, SA, LCP)
+
+
+def compute_cis_lpf(n, SA, LCP):
```

Comment:
Nit: I think in e.g. lcp.py et al. there's an informal convention to use (SA, LCP, n) as parameters

Reply:
Ack